tests/unit/test_parsers.py

```diff
@@ -1622,6 +1622,97 @@ def test_can_parse_route53_with_missing_message(self):
         self.assertEqual(error['Message'], '')
 
 
+class TestS3ExpiresResponses(unittest.TestCase):
+    def setUp(self):
+        self.parser = parsers.S3Parser()
+        self.output_shape = model.StructureShape(
+            'OutputShape',
+            {
+                'type': 'structure',
+                'members': {
+                    'Expires': {
+                        'shape': 'Expires',
+                        'location': 'header',
+                        'locationName': 'Expires',
+                    }
+                },
+            },
+            model.ShapeResolver({'Expires': {'type': 'timestamp'}}),
+        )
+
+    def test_valid_expires_response_parsed(self):
+        expires_value = 'Thu, 01 Jan 2015 00:00:00 GMT'
+        parsed = self.parser.parse(
+            {
+                'headers': {'Expires': expires_value},
+                'body': b'',
+                'status_code': 200,
+            },
+            self.output_shape,
+        )
+        expected = {
+            'ResponseMetadata': {
+                'HTTPStatusCode': 200,
+                'HTTPHeaders': {'expires': expires_value},
+            },
+            'Expires': datetime.datetime(2015, 1, 1, tzinfo=tzutc()),
+            'ExpiresString': expires_value,
+        }
+        self.assertEqual(parsed, expected)
+
+    def test_invalid_expires_response_parsed(self):
```

Comment:
We should be testing all these known invalid timestamp values:
```
2034-01-01T00:00:00Z
2023-11-02 08:43:04 -0400
Invalid Date
access plus 1 month
2023-10-12 23:00:00 UTC
Thu, 01 Jan 1970 00:00:00 UTC
Tue, 15 Apr 2014 07:17:09 UTC
10/21/2018
Expires: Thu, 9 Sep 2013 14:19:41 GMT
01 dec 2100
Sun, 31 Aug 2031 20:56:53 GMT+00:00
Sun, 22 Oct 23 00:45:02 UTC
access plus 1 month
{ts '2023-10-10 09:27:14'}
Tue, 01 Jan 2014 12:00:00 GMT
Sat, 10 Jan 2049 00:00:00 GMT
```

Reply:
The thing is not all of these are considered invalid in the SDK. As long as it's in a timestamp format, our parsers are able to parse them as such. For example, `Tue, 01 Jan 2014 12:00:00 GMT` is "invalid" in this list because 01/01/14 was on a Wednesday instead of a Tuesday.
However, I can add the ones which are not parsable as a timestamp to the test.